mesonbuild/dependencies/ui.py

```diff
@@ -45,6 +45,11 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
             self.link_args = ['-framework', 'OpenGL']
             # FIXME: Detect version using self.clib_compiler
             return
+        if self.env.machines[self.for_machine].is_linux():
```

Comment:
We could either explicitly check for linux, or we could convert this to if/elif/else and do the find_library trick on all systems other than darwin or Windows. What are your thoughts?

Reply:
I've switched this round to make this happen. Darwin and Windows are handled first.
I couldn't figure out how to detect the version, so the output remains `Run-time dependency gl found: YES` with no version number. I assume it's just not possible to drag it out from `clib_compiler` anyway, and it's a disadvantage of not relying on `pkg-config`, but I could have missed something.

Reply:
That makes sense to me too, yeah.